Bluetooth: Channel Sounding: Add `bt_cs_set_default_settings`

sean-madigan opened the pull request:
Add CS set default settings command for the host, along with kconfig and shell support.

sean-madigan committed:
bluetooth: shell: Add CS to BT shell with set default settings command

Use a new file and command for this where all CS commands
can live.

Added support for bt_cs_set_default_settings command.

Added test case where shell is built with CS to ensure code
is built in CI.

Signed-off-by: Sean Madigan <sean.madigan@nordicsemi.no>

sean-madigan committed:
bluetooth: kconfig: Add channel sounding kconfigs

Add new controller and host kconfigs for Bluetooth 6.0
Channel Sounding.

Signed-off-by: Sean Madigan <sean.madigan@nordicsemi.no>

olivier-le-sage commented on subsys/bluetooth/host/cs.c:
```diff
@@ -0,0 +1,49 @@
+	cp->role_enable = 0;
+
+	if (params->enable_initiator_role) {
+		cp->role_enable &= BT_HCI_OP_LE_CS_INITIATOR_ROLE_MASK;
```
shouldn't this be |= ?

sean-madigan replied:
of course

olivier-le-sage commented on subsys/bluetooth/host/cs.c:
```diff
@@ -0,0 +1,49 @@
+	struct bt_hci_cp_le_cs_set_default_settings *cp;
+	struct net_buf *buf;
+
+	if (params->max_tx_power > BT_HCI_OP_LE_CS_MAX_MAX_TX_POWER ||
```
I think the controller also has to check this, and returns "invalid command parameters" in this case. So we could choose to leave this out

sean-madigan replied:
yeah good point, no need to duplicate the logic

olivier-le-sage commented:
I would slightly prefer channel_sounding.c and channel_sounding.h over acronyms.

I think that's a bit nicer for people who are less familiar, or browsing the code casually

Feel free to leave it as is

sean-madigan commented:
> I would slightly prefer channel_sounding.c and channel_sounding.h over acronyms.
> 
> I think that's a bit nicer for people who are less familiar, or browsing the code casually
> 
> Feel free to leave it as is

I may leave as is, I think cs will be a useful acronym to use

theob-pro commented on subsys/bluetooth/shell/cs.c:
```diff
@@ -0,0 +1,123 @@
+		set_default_settings, NULL,
+		"<Enable initiator role: true, false> <Enable reflector role: true, false> "
+		" <CS_SYNC antenna selection: 0x01 - 0x04, 0xFE, 0xFF> <Max TX power: -127 - 20>",
+		cmd_set_default_settings, 1, 4),
```
Shouldn't it be as follow? The arguments are mandatory right?

```suggestion
		cmd_set_default_settings, 5, 0),
```

sean-madigan replied:
Good point

theob-pro commented on tests/bluetooth/shell/testcase.yaml:
```diff
@@ -43,6 +43,13 @@ tests:
+      - CONFIG_BT_CHANNEL_SOUNDING=y
+      - CONFIG_BT_CTLR=n
+    platform_allow:
+      - native_posix
```
Please use `native_sim`. `native_posix` is deprecated: https://docs.zephyrproject.org/latest/boards/native/native_posix/doc/index.html

```suggestion
      - native_sim
```

olivier-le-sage commented on include/zephyr/bluetooth/cs.h:
```diff
@@ -0,0 +1,86 @@
+	BT_CS_ANTENNA_SELECTION_OPT_NO_RECOMMENDATION = BT_HCI_OP_LE_CS_ANTENNA_SEL_NONE,
+};
+
+/** Connection subrating parameters for LE connections */
```
noticed this now

suggest copying some text from the core spec, like "default CS
settings in the local Controller"

olivier-le-sage commented on subsys/bluetooth/host/cs.c:
```diff
@@ -0,0 +1,44 @@
+	}
+
+	if (params->enable_reflector_role) {
+		cp->role_enable |= BT_HCI_OP_LE_CS_INITIATOR_ROLE_MASK;
```
should be bit 1, BT_HCI_OP_LE_CS_REFLECTOR_ROLE_MASK

sean-madigan committed:
bluetooth: host: Add support for CS set default settings

Added support for a new API for setting default channel
sounding settings, this is mainly a wrapper around the HCI
command.

For this add a new module for channel sounding, where new
channel sounding APIs will go.

Signed-off-by: Sean Madigan <sean.madigan@nordicsemi.no>
